packages/text-input/src/TextInput.spec.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import {
+  render,
+  cleanup,
+  fireEvent,
+  getByTitle,
+  getByText,
+} from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import TextInput, { State } from './TextInput';
+import { typeIs } from '@leafygreen-ui/lib';
+
+afterAll(cleanup);
+
+describe('packages/text-input', () => {
+  const className = 'test-text-input-class';
+  const label = 'Test Input Label';
+  const description = 'This is the description';
+  const error = 'This is the error message';
+  const validEmail = 'test.email@mongodb.com';
+  const invalidEmail = 'invalid.email';
+  const onChange = jest.fn();
+  onChange.mockReturnValue('none');
+
+  const renderedTextInput = render(
+    <TextInput
+      className={className}
+      label={label}
+      description={description}
+      onChange={onChange}
+      state={State.none}
+      optional={true}
+    />,
+  );
+
+  const renderedChildren = renderedTextInput.container.firstChild;
+
+  if (!typeIs.element(renderedChildren)) {
+    throw new Error('TextInput component failed to render');
+  }
+
+  const renderedInputElement =
```

Comment:
We're trying to move away form relying on the order of the DOM to find our components when testing. What might make that easier for you is to try grabbing getByText and getByTestId methods from the container when you render.

`const {getByText, getByTestId} = render(<TextInput />)`

This should allow you to look for your component more directly, without having to do as much DOM traversal.

Reply:
Fixed